src/modules/mavlink/mavlink_shell.cpp

```diff
@@ -113,6 +115,17 @@ int MavlinkShell::start()
 	fflush(stdout);
 	fflush(stderr);
 
+#ifdef __PX4_POSIX
+	int remote_in_fd = dup(_shell_fds[0]);	// Input file descriptor for the remote shell
```

Comment:
Is `dup` required here?

Reply:
Yes.  Otherwise the close() below would need to be #ifdef with POSIX.  I believe this is cleaner.